File: lib/models/madde_data.dart
class MaddeData {
  final String maddeName;
  final double changeRate;

  MaddeData({
    required this.maddeName,
    required this.changeRate,
  });

  factory MaddeData.fromCsv(List<dynamic> row, int columnIndex) {
    return MaddeData(
      maddeName: row[1].toString().trim(),
      changeRate: double.tryParse(row[columnIndex].toString()) ?? 0.0,
    );
  }
}

File: src/models/MaddeData.ts
export interface MaddeData {
  maddeName: string;
  changeRate: number;
}

export class MaddeDataModel implements MaddeData {
  maddeName: string;
  changeRate: number;

  constructor(maddeName: string, changeRate: number) {
    this.maddeName = maddeName;
    this.changeRate = changeRate;
  }

  static fromCsv(row: any[], columnIndex: number): MaddeData {
    return new MaddeDataModel(
      row[1]?.toString().trim() || '',
      parseFloat(row[columnIndex]?.toString() || '0') || 0.0
    );
  }
}

